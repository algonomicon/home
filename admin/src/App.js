import { Component } from 'inferno'
import { BrowserRouter } from 'inferno-router'
import { Provider as StoreProvider } from 'inferno-mobx'
import { Provider as RenderProvider } from 'inferno-fela'
import { ThemeProvider } from 'inferno-fela'
import { createRenderer } from 'fela'
import Routes from './Config/Routes'
import Stores from './Config/Stores'
import GlobalStyles from './Static/Assets/GlobalStyles.css'

const renderer = createRenderer()
renderer.renderStatic(GlobalStyles.toString())

const theme = {
  hufflepuff: '#9370DB',
  ravenclaw: '#5F527A',
  gryffindor: '#FDCA7A'
}

class App extends Component {
  render() {
    return (
      <RenderProvider renderer={renderer}>
        <ThemeProvider theme={theme}>
          <StoreProvider {...Stores}>
            <BrowserRouter>
              {Routes}
            </BrowserRouter>
          </StoreProvider>
        </ThemeProvider> 
      </RenderProvider>
    )
  }
}

export default App