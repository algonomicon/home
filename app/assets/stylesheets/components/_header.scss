header {
  background:url(/assets/nebula.jpg) no-repeat center;
  background-size:cover;
  height:35vh;
  transition:all 1s ease;

  .wrapper {
    position:relative;
    height:100%;

    .masthead {
      position:relative;
      margin-top:10px;
      display:flex;
      align-items:center;
      justify-content:space-between;

      .logo {
        a {
          display:block;

          svg {
            fill:white;
            max-height:25px;
            width:auto;
            shape-rendering: crispedges;
            transition:all 0.3s ease;
          }

          &:hover {
            svg {
              opacity:1;
            }
          }
        }
      }
      
      nav {
        display:flex;
        align-items:center;

        a {
          color:white;
          margin:0 10px;

          &:hover {
            color:$rust1;
          }
        }
        span {
          color:white;
          margin:0 5px;
          font-weight:bold;
          cursor:default;
        }
      }
    }

    .intro {
      position:absolute;
      width:100%;
      left:10px; bottom:20px;

      .tagline {
        color:white;
        @include font-size(14);

      }
      .title {
        color:white;
        margin-bottom:10px;
      }
      .search {
        display:flex;

        .input-group {
          position:relative;
          width:100%; max-width:400px;

          .input-icon {
            position:absolute;
            left:15px; top:50%;
            transform:translateY(-50%);

            i {
              @include font-size(20);
              color:$light3;
            }
          }

          input {
            width:100%;
            padding:20px;
            padding-left:50px;
            border-radius:3px;
          }
        }
        .search-button {
          display:block;
          color:white;
          background:linear-gradient(to left, $rust1, $rust2);
          padding:20px 40px;
          border-radius:3px;
          margin:0 10px;
        }
      }
    }
  }

  &.grayscale {
    filter:grayscale(1);
    transition:all 3s ease;
  }
}