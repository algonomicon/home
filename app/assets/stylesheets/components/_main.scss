body {
  transition:all 1s ease;
  
  &.grayscale {
    filter:grayscale(1);
    transition:all 3s ease;
  }

  main {
    min-height:60vh;
    padding-bottom:40px;
  }
}