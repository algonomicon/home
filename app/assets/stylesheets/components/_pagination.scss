.pagination {
  padding:5px 0;
  margin-left:-5px;
  
  span {
    a {
      color:$light3;
      padding:5px;

      &:hover {
        color:$purple2;
      }
    }
    &.current {
      padding:5px;
      color:$dark1;
      font-weight:bold;
    }
  }
}