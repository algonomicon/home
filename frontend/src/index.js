import { render } from 'inferno'
import { BrowserRouter } from 'inferno-router'
import { Provider as StoreProvider } from 'inferno-mobx'
import { Provider as RenderProvider } from 'inferno-fela'
import { createRenderer } from 'fela'
import GlobalStyles from './Static/Assets/GlobalStyles.css'
import Routes from './Config/Routes'
import Stores from './Config/Stores'
import './Static/Assets/AppStyles.scss'
import * as serviceWorker from './serviceWorker'


const renderer = createRenderer()
renderer.renderStatic(GlobalStyles.toString())

render((
  <RenderProvider renderer={renderer}>
    <StoreProvider {...Stores}>
      <BrowserRouter>
        {Routes}
      </BrowserRouter>
    </StoreProvider>
  </RenderProvider>
), document.getElementById('root'))

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister()
