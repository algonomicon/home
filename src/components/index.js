export { default as Categories } from "./categories"
export { default as Feed, FeedQuery } from "./feed"
export { default as Footer } from "./footer"
export { default as Header } from "./header"
export { default as Layout } from "./layout"
export { default as Logo } from "./logo"
export { default as Main } from "./main"
export { default as Minion } from "./minion"
export { default as Search } from "./search"
export { default as Sidebar } from "./sidebar"
export { default as Sort } from "./sort"
export { default as Tags } from "./tags"
export { default as Wrapper } from "./wrapper"
