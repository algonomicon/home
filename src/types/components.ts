export type FeedProps = {
  data: {
    allMarkdownRemark: {
      nodes: {
        excerpt: string
        frontmatter: {
          title: string
          slug: string
          date: string
        }
        fields: {
          collection: string
        }
      }[]
    }
  }
}

export type FeedItemProps = {
  data: {
    excerpt: string
    frontmatter: {
      title: string
      slug: string
      date: string
    }
    fields: {
      collection: string
    }
  }
}
