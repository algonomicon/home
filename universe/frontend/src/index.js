import { render } from 'inferno'
import { createRenderer } from 'fela'
import { Provider } from 'inferno-fela'
import App from './App/App'
import GlobalStyles from './App/GlobalStyles.css'
import * as serviceWorker from './serviceWorker'

const renderer = createRenderer()
renderer.renderStatic(GlobalStyles.toString())

render((
  <Provider renderer={renderer}>
    <App />
  </Provider>
), document.getElementById('root'))

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister()
