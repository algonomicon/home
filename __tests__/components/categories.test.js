import React from "react"
import renderer from "react-test-renderer"

import { Categories } from "../../src/components"

describe("Categories", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Categories />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
