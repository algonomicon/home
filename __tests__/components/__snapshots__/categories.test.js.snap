// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders correctly 1`] = `
<div
  className="sc-bdVaJa knWkmq"
>
  <a
    className="sc-bwzfXH FtMCK"
    href="/"
  >
    Home
  </a>
  <a
    className="sc-bwzfXH FtMCK"
    href="/articles"
    partiallyActive={true}
  >
    Articles
  </a>
  <a
    className="sc-bwzfXH FtMCK"
    href="/datasets"
    partiallyActive={true}
  >
    Datasets
  </a>
  <a
    className="sc-bwzfXH FtMCK"
    href="/papers"
    partiallyActive={true}
  >
    Papers
  </a>
  <a
    className="sc-bwzfXH FtMCK"
    href="/projects"
    partiallyActive={true}
  >
    Projects
  </a>
  <a
    className="sc-bwzfXH FtMCK"
    href="/snippets"
    partiallyActive={true}
  >
    Snippets
  </a>
</div>
`;
